src/mongocxx/exception/authentication_exception.hpp

```diff
@@ -16,50 +16,43 @@
 
 #include <mongocxx/config/prelude.hpp>
 
-#include <cstdint>
-#include <exception>
-#include <tuple>
-
 #include <bsoncxx/document/value.hpp>
-#include <bsoncxx/document/view.hpp>
 #include <bsoncxx/stdx/optional.hpp>
-
+#include <mongocxx/exception/exception.hpp>
 #include <mongocxx/stdx.hpp>
 
 namespace mongocxx {
 MONGOCXX_INLINE_NAMESPACE_BEGIN
-namespace exception {
-
-using error_and_code_type = std::tuple<std::string, std::int32_t>;
 
-class MONGOCXX_API base : virtual public std::exception {
+class MONGOCXX_API authentication_exception : public exception {
```

Comment:
If you wanted to handle them differently, you could always do

```
try {
   // do stuff
}
catch (const authentication_exception&) {
    // We handle auth issues at a higher level, just
    // let the exception through
    throw;
} catch (const operation_exception& op_xcp) {
    // OK, I actually want to do something here, this isn't an auth issue.
}
```

But if you *do* want to handle auth exceptions the same way, then you can just catch `operation_exception`. So I think it can just be a subclass of operation_exception? @samantharitter what do you think?